### PR TITLE
Switch repo to use GHA, deploy to NPM and work with Node v18

### PR DESCRIPTION
**Pull request by prajapac**
- Also, update Readme with relevant update information

Part of work to update dependencies (specifically `node-sass`) for:
[US151517](https://rally1.rallydev.com/#/289692574792d/custom/355050439968?Iteration=u&Release=u&c_UXDevKanbanStates=Defined&cpoid=289692574792&detail=%2Fuserstory%2F698104033025&rankScope=BACKLOG&typeDef=7362609683) - Upgrade the Node version used in Lessons to v18 from v14

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Release
+on:
+  push:
+    branches:
+      - master
+jobs:
+  release:
+    name: Release
+    timeout-minutes: 5
+    runs-on: [self-hosted, Linux, AWS]
```
Comment (prajapac) [inline .github/workflows/release.yml:10]: I'm not sure what exactly `self-hosted` means in our context, or what the `AWS` tag means. [Lessons](https://github.com/Brightspace/smart-curriculum/blob/master/.github/workflows/ci.yml#L7) uses this, while other repos use something like `runs-on: ubuntu-latest`.kflows/ci.yml#L7) uses this, while other repos use something like `runs-on: ubuntu-latest`.

Comment (dlockhart) [reply .github/workflows/release.yml:10]: You want `runs-on: ubuntu-latest` for public repos like this one.

Comment (dlockhart) [reply .github/workflows/release.yml:10]: You want `runs-on: ubuntu-latest` for public repos like this one.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          persist-credentials: false
+      - name: Setup Node
+        uses: Brightspace/third-party-actions@actions/setup-node
+        with:
+            node-version-file: .nvmrc
+      - name: Install dependencies
+        run: npm install
+      - name: Lint and Test
+        run: npm run test
```
[the remark this answers is not in the document]
Comment (dlockhart) [reply .github/workflows/release.yml:23]: Typically we don't re-run the test steps on `main`/`master`.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Semantic Release
+        uses: BrightspaceUI/actions/semantic-release@master
+        with:
+          DEFAULT_BRANCH: master
+          GITHUB_TOKEN: ${{ secrets.D2L_GITHUB_TOKEN }}
+          NPM: true
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
[the remark this answers is not in the document]
Comment (dlockhart) [reply .github/workflows/release.yml:30]: I've just set them up for you!

**Comment by dlockhart** [comment]
Due to the requirement that this uses Node 18, when this merges I think you'll want to do it as a major release (breaking change). There are 5 other projects using version `1.0.0` and the rest are still stuck on `0.0.3`, so likely not a huge deal to upgrade those 5 and something you likely already need to do anyway.

I can help ensure it does a major release -- I can merge it for you if you like.

File: .github/workflows/release.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (prajapac) [reply .github/workflows/release.yml:23]: Gotcha, I'll fix in next commit!

File: .github/workflows/release.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (prajapac) [reply .github/workflows/release.yml:30]: Thanks! 😄

File: .github/workflows/release.yml (hunk 1 above)
Comment (prajapac) [reply .github/workflows/release.yml:10]: Gotcha, I'll fix in next commit!

**Comment by prajapac** [comment]
> Due to the requirement that this uses Node 18, when this merges I think you'll want to do it as a major release (breaking change). There are 5 other projects using version `1.0.0` and the rest are still stuck on `0.0.3`, so likely not a huge deal to upgrade those 5 and something you likely already need to do anyway.
> 
> I can help ensure it does a major release -- I can merge it for you if you like.

I figured as much, and was planning to do so, adding `BREAKING CHANGE:` to the commit message should do the trick eh? I'll be honest, I'm excited to do my first major version bump lol, but you can do the merge too if that's preferable 😺

**Comment by dlockhart** [comment]
Go for it! Make sure you do a squash commit and put "BREAKING CHANGE:" (with the colon) in both the squash commit message and the extended description -- the extended description is the most important I believe.

Screenshot 2023-05-03 at 3 27 47 PM

**Review by dlockhart — APPROVED** [review]

**Pull request merged**